src/AsmResolver/IO/ZeroesDataSource.cs

```diff
@@ -54,5 +57,15 @@ public int ReadBytes(ulong address, byte[] buffer, int index, int count)
             Array.Clear(buffer, index, count);
             return actualLength;
         }
+
+#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP2_1_OR_GREATER
+        /// <inheritdoc />
+        public int ReadBytes(ulong address, Span<byte> buffer)
+        {
+            int actualLength = (int) Math.Min(Length, (uint) buffer.Length);
+            buffer.Clear();
```

Comment:
The code here is clearing the entire input buffer, which matches the current behavior of the corresponding `byte[]` overload. However, this seems like a performance trap if a large buffer is passed in, since technically only `[0..actualLength)` needs to be cleared.

Was this intentional behavior? If not, I could change this to `buffer[..actualLength].Clear();` and change the `byte[]` overload to use `Array.Clear(buffer, index, actualLength);` to avoid doing more work in the method than necessary.

Reply:
This is an oversight. Your proposed change makes sense, also for the original API. If you can change it that would be great!